src/main/java/randoop/main/CoverageTracker.java

```diff
@@ -0,0 +1,192 @@
+package randoop.main;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import org.jacoco.core.analysis.*;
+import org.jacoco.core.data.ExecutionDataStore;
+import org.jacoco.core.data.SessionInfoStore;
+import org.jacoco.core.instr.Instrumenter;
+import org.jacoco.core.runtime.IRuntime;
+import org.jacoco.core.runtime.LoggerRuntime;
+import org.jacoco.core.runtime.RuntimeData;
+
+/**
+ * Tracks the coverage of each method under test. Largely based on
+ * http://www.jacoco.org/jacoco/trunk/doc/examples/java/CoreTutorial.java
+ */
+public class CoverageTracker {
+  public static final CoverageTracker instance = new CoverageTracker();
+
+  private final IRuntime runtime = new LoggerRuntime();
+  private final MemoryClassLoader memoryClassLoader = new MemoryClassLoader();
+
+  private final ExecutionDataStore executionData = new ExecutionDataStore();
+  private final SessionInfoStore sessionInfos = new SessionInfoStore();
+
+  private final Instrumenter instrumenter;
+  private final RuntimeData data;
+
+  // Map from method name to coverage details.
+  private final Map<String, CoverageDetails> coverageDetailsMap = new HashMap<>();
+
+  // Map for all classes that are instrumented through the {@code CoverageTracker} class.
+  private final Map<String, Class<?>> instrumentedClasses = new HashMap<>();
+
+  /**
+   * Coverage details related to a single method under test. Tracks total number of branches and
+   * number of uncovered branches.
+   */
+  public static class CoverageDetails {
+    private int numBranches;
+    private int uncoveredBranches;
+
+    private void setNumBranches(int numBranches) {
+      this.numBranches = numBranches;
+    }
+
+    private void setUncoveredBranches(int uncoveredBranches) {
+      this.uncoveredBranches = uncoveredBranches;
+    }
+
+    public int getNumBranches() {
+      return numBranches;
+    }
+
+    public int getUncoveredBranches() {
+      return uncoveredBranches;
+    }
+  }
+
+  /** Initializes the coverage tracker. */
+  private CoverageTracker() {
+    instrumenter = new Instrumenter(runtime);
+
+    data = new RuntimeData();
+    try {
+      runtime.startup(data);
+    } catch (Exception e) {
+      e.printStackTrace();
```

Comment:
Should this terminate execution too?  Otherwise it's likely that a user won't notice the logging.